src/kbc_modules/cc_kbc/attester/sev_snp_vtpm/mod.rs

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::Attester;
+use anyhow::*;
+use serde::{Deserialize, Serialize};
+use vtpm_snp::hcl::HclReportWithRuntimeData;
+use vtpm_snp::vtpm;
+
+pub fn detect_platform() -> bool {
+    vtpm::has_tpm_device()
+}
+
+#[derive(Debug, Default)]
+pub struct VtpmAttester;
+
+#[derive(Serialize, Deserialize)]
+struct VtpmSnpEvidence {
+    quote: vtpm::Quote,
+    hcl_report: HclReportWithRuntimeData,
+}
+
+impl Attester for VtpmAttester {
+    fn get_evidence(&self, report_data: String) -> Result<String> {
```

Comment:
How is someone consuming this API getting the `report_data`? IIUC they would read it from the `0x01400001` right?

Reply:
That would be report_data provided as an input, prior to generating a report. It's the challenge part of the [request-challenge-attestation-response](https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#request-challenge-attestation-response-semantics) flow.

In a common sev-snp scenario, this would be an input arg to the IOCTL api and end up in the SNP's report_data field. In the HCL case there's more redirection and report layers, but we'd still verify this report_data in cc/attestation-service as nonce via the `vtpm::verify_quote()` api.